tests/unit_tests/encoder/categorical/test_autoencoder.py

```diff
@@ -46,3 +46,28 @@ def test_autoencoder(self):
         encoder_accuracy = accuracy_score(list(map(str, test_data)), list(map(str, decoded_data)))
         print(f'Categorical encoder accuracy for: {encoder_accuracy} on testing dataset')
         self.assertTrue(encoder_accuracy > 0.70)
+
+    def check_encoder_on_device(self, device):
+        enc = CategoricalAutoEncoder(stop_after=5, device=device)
```

Comment:
A lot of this test seems to be copy-pasted from the one above. Could you abstract the data generation procedure into a common function that gets called by both?

(Note: I am ignoring the fact that this way of generating data seems very much unoptimal, but then again it wasn't introduced in this PR and it's also not its objective, so we will tackle it in another issue).